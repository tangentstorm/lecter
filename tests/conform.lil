# conforming between atoms, lists, and dictionaries.

# monadic:
show[floor (55.3)]
show[floor (1.1,2.2,-3.3)]
show[floor ("A","B") dict 11.2,33.9]
show[unit  "ABCD" dict pi*(0,.5,1,1.5,2)]

# dyadic, the basics:
show[11+22]
show[100+(22,33)]
show[(11,22)+100]
show[(11,22)+(100,200)]

# jagged lists keep the left arg's length:
show[(11,22,33)+(100,200)]
show[(11,22)+(100,200,300)]

# dict-dict takes the union of keys:
x:("White","Brown","Speckled")dict 10,34,27
y:("White","Speckled","Brown")dict  5, 3, 8
z:("Brown","White","Blue"    )dict  9,13,35
show[x+y] # prefer left order
show[x+z] # zero-fill unmatched

# atom-dict spreads the atom to every dict element:
show[x+100]
show[200+x]

# dict-nondict / nondict-dict likewise spreads:
d:("alpha","beta")dict (list 11,22),(list 33,44) 
show[(100,200)*d]
show[d*(100,200)]

# reducers reflect dict conforming
show[sum (list x),(list y),(list z)]
