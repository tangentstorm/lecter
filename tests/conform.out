55
(1,2,-4)
{"A":11,"B":33}
{"A":(1,0),"B":(0,1),"C":(-1,0),"D":(-0,-1)}
33
(122,133)
(111,122)
(111,222)
(111,222,133)
(111,222)
{"White":15,"Brown":42,"Speckled":30}
{"White":23,"Brown":43,"Speckled":27,"Blue":35}
{"White":110,"Brown":134,"Speckled":127}
{"White":210,"Brown":234,"Speckled":227}
{"alpha":(1100,4400),"beta":(3300,8800)}
{"alpha":(1100,4400),"beta":(3300,8800)}
{"White":28,"Brown":51,"Speckled":30,"Blue":35}
